examples/trials/weight_sharing/ga_squad/evaluate.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge,
+# to any person obtaining a copy of this software and associated
+# documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+'''
+Evaluation scripts for QA model.
+'''
+
+from __future__ import print_function
+from collections import Counter
+import string
+import re
+import argparse
+import json
+import sys
+
+
+def normalize_answer(str_input):
+    """Lower text and remove punctuation, articles and extra whitespace."""
+    def remove_articles(text):
+        '''
+        Remove "a|an|the"
+        '''
+        return re.sub(r'\b(a|an|the)\b', ' ', text)
+
+    def white_space_fix(text):
+        '''
+        Remove unnessary whitespace
+        '''
+        return ' '.join(text.split())
+
+    def remove_punc(text):
+        '''
+        Remove punc
+        '''
+        exclude = set(string.punctuation)
+        return ''.join(ch for ch in text if ch not in exclude)
+
+    def lower(text):
+        '''
+        Change string to lower form.
+        '''
+        return text.lower()
+
+    return white_space_fix(remove_articles(remove_punc(lower(str_input))))
+
+
+def f1_score(prediction, ground_truth):
+    '''
+    Calculate the f1 score.
+    '''
+    prediction_tokens = normalize_answer(prediction).split()
+    ground_truth_tokens = normalize_answer(ground_truth).split()
+    common = Counter(prediction_tokens) & Counter(ground_truth_tokens)
+    num_same = sum(common.values())
+    if num_same == 0:
+        return 0
+    precision = 1.0 * num_same / len(prediction_tokens)
+    recall = 1.0 * num_same / len(ground_truth_tokens)
+    f1_result = (2 * precision * recall) / (precision + recall)
```

Comment:
1. This could cause divide zero error.  Common practice is to add a small number on denominator.
2. Need to check len(prediction_tokens) and len(ground_truth_tokens) not zero.